core/src/reporter/utils.ts

```diff
@@ -251,3 +254,30 @@ export async function sendTransactionCaver({
     throw new OraklError(OraklErrorCode.CaverTxTransactionFailed)
   }
 }
+
+async function extractSubmissionData(input: string) {
+  const iface = new ethers.utils.Interface(Aggregator__factory.abi)
+
+  const [, submission] = iface.decodeFunctionData('submit', input)
+  return Number(submission)
+}
+
+export async function makeSubmissionData({
+  to,
+  payload,
+  logger
+}: {
+  to: string
+  payload: string
+  logger?: Logger
+}) {
+  const aggregator = await loadAggregatorByAddress({ address: to, logger })
+  const aggregatorId = Number(aggregator.id)
```

Comment:
ID of aggregator is stored in DB as `BigInt`, therefore we should not downcast it for `Number`. For now, the values are small, and most likely will never reach values above `Number` range, but I think it makes more sense to keep it in the correct data format rather than converting it to `Number`.